### DISCUSSION TITLE
Part 2

### DISCUSSION THREAD

**Discussion by LayZeeDK** [this-is-learning/rxjs-fundamentals-course#2] [Content contributions]
Discussing and coordinating [Part 2](https://github.com/this-is-learning/rxjs-fundamentals-course/blob/main/part-2/part-2.md) efforts.

**Comment by LayZeeDK** [comment]
# Contributors
@aboudard is actively looking into this part 🙇‍♂️

**Reply by LayZeeDK** [reply]
@mainawycliffe is joining this effort 🙌

**Comment by aboudard** [comment]
Part 1 : Declarative and Imperative Programming

> Solution 2 is a little more interesting. x is an array, so we know that this code is synchronous.

Maybe we refer to the Array map() presentation : no element added to the array after the call will be processed.

> Another thing to note about Solution 2 is that it is composable, meaning we could chain or pipe a sequence of operations together.

Maybe can you start saying that map() does return a new Array, and introduce like that the idea of chaining / being composable.

**Comment by aboudard** [comment]
Part 2 : Reactive programming

Yes, I think this needs some content, I would go with examples on how Javascript has in itself native asynchronous streams of values, and how indeed combine them to reach a specific functional goal cam quickly lead to callback hell.

I'll start writing some content / simple code.

**Reply by LayZeeDK** [reply]
@aboudard
It seems that more content was planned. See [the outline](https://github.com/this-is-learning/rxjs-fundamentals-course/blob/main/notes/outline.md) and the Part 2 table of contents.

**Reply by aboudard** [reply]
Ah ok, I didn't see this.
So, we follow the outline ?

**Reply by LayZeeDK** [reply]
Unless you have a better idea 🙂 It's not a requirement though.

**Comment by mainawycliffe** [comment]
I was thinking of working on some sections on these parts, not sure where @aboudard is at with it - specifically the reactive programming section.

**Reply by LayZeeDK** [reply]
@mainawycliffe
Try reaching out in the Discord server to coordinate 🙂
